app/services/structured_fact_importer.rb

```diff
@@ -19,6 +19,28 @@ def normalize_recurse(memo, facts, prefix = '')
     memo
   end
 
+  def find_or_create_fact_name(name, value)
+    if name.include?(FactName::SEPARATOR)
```

Comment:
What is this doing? It tries to find who is the parent fact or create it when the factname has a separator?  
Extracting to a separate method with a name that explains what it does and/or comments would help here IMO